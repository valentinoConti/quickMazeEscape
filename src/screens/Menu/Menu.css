#menuScreen {
  position: absolute;
  width: 100%;
  height: 100%;
  background: url('../../../assets/menu.png');
  outline: black 2px solid;
  font-size: 24px;
  text-align: center;
}

footer {
  position: absolute;
  bottom: 10px;
  left: 15px;
  font-size: 24px;
  text-shadow: 2px 2px 4px black;
  color: black;
}

.menuTitle {
  text-align: right;
  margin-right: 100px;
  color: black;
  text-shadow: 1px 1px 2px black;
}

.menuButton {
  margin-left: 280px;
  width: 220px;
  height: 40px;
  background-color: white;
  border: 2px solid black;
  box-shadow: 2px 2px 2px black;
  cursor: pointer;
  font-size: 18px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}