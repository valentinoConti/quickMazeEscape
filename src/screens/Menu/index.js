import React from 'react';
import PropTypes from 'prop-types';
import './Menu.css';

/**
 * @name Menu
 * @description Menu screen. Here we have our different screens links (Play, Instructions and Fullscreen toggle)
 * @param {Function} setScreen 
 */
const Menu = ({setScreen}) => {
  const playPressed = () => setScreen('play');
  const controlsPressed = () => setScreen('controls');
  const toggleFullscreen = () => {
    const doc = document.documentElement;
    if (doc.requestFullscreen) {
      doc.requestFullscreen();

      let scale = 0.9;
      while (
        document.querySelector('#root').getBoundingClientRect().height+75 < window.innerHeight &&
        document.querySelector('#root').getBoundingClientRect().width < window.innerWidth
      ) {
        scale += 0.025;
        document.querySelector('#root').style.transform = `scale(${scale})`;
      }
    }
  };
  
  return (
    <div id="menuScreen">
      <h1 className="menuTitle">Quick-Maze Escape</h1>

      <button
        id="playButton"
        className="menuButton"
        onClick={playPressed}>
        Play
      </button>

      <button
        id="instructionsButton"
        className="menuButton marginTop"
        onClick={controlsPressed}>
        Instructions
      </button>

      <button
        id="fullscreenButton"
        className="menuButton marginTop"
        onClick={toggleFullscreen}>
        Fullscreen
      </button>

      <footer>
        Valentino Conti
      </footer>
    </div>
  );
};

Menu.propTypes = {
  setScreen: PropTypes.func.isRequired
};

export default Menu;
