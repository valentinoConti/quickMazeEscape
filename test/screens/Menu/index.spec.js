import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import Menu from '../../../src/screens/Menu';

configure({adapter: new Adapter()});

describe('Menu screen test', () => {
  // FullScreen functionality mocks
  document.requestFullscreen = jest.fn();
  document.documentElement.requestFullscreen = jest.fn();
  window.innerHeight = 700;
  window.innerWidth = 700;
  var scale = 400;
  document.querySelector = jest.fn(() => ({
    getBoundingClientRect: () => { scale += 100; return({width: scale, height: scale}) },
    style: { transform: ()=>{} }
  }))

  const mockSetScreen = jest.fn();
  let wrapper;

  it('should take/compare snapshot', () => {
    wrapper = mount(
      <Menu setScreen={mockSetScreen} />
    );
    expect(wrapper.find('#menuScreen')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test the playPressed, controlsPressed & toggleFullscreen functions', () => {
    expect(mockSetScreen).not.toHaveBeenCalled();
    wrapper.find('button#playButton').simulate('click');
    expect(mockSetScreen).toHaveBeenCalledWith('play');

    wrapper.find('button#instructionsButton').simulate('click');
    expect(mockSetScreen).toHaveBeenCalledWith('controls');

    wrapper.find('button#fullscreenButton').simulate('click');
    expect(document.documentElement.requestFullscreen).toHaveBeenCalled();
  });
});