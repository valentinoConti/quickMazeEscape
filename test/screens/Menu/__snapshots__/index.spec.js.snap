// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu screen test should take/compare snapshot 1`] = `
<Menu
  setScreen={[MockFunction]}
>
  <div
    id="menuScreen"
  >
    <h1
      className="menuTitle"
    >
      Quick-Maze Escape
    </h1>
    <button
      className="menuButton"
      id="playButton"
      onClick={[Function]}
    >
      Play
    </button>
    <button
      className="menuButton marginTop"
      id="instructionsButton"
      onClick={[Function]}
    >
      Instructions
    </button>
    <button
      className="menuButton marginTop"
      id="fullscreenButton"
      onClick={[Function]}
    >
      Fullscreen
    </button>
    <footer>
      Valentino Conti
    </footer>
  </div>
</Menu>
`;
